src/shared_modules/content_manager/src/components/executionContext.hpp

```diff
@@ -46,18 +47,39 @@ class ExecutionContext final : public AbstractHandler<std::shared_ptr<UpdaterBas
         unsigned int databaseOffset;
         try
         {
-            databaseOffset = std::stoi(context.spRocksDB->getLastKeyValue().second.ToString());
+            databaseOffset =
+                std::stoi(context.spRocksDB->getLastKeyValue(Components::Columns::CURRENT_OFFSET).second.ToString());
         }
         catch (const std::runtime_error&)
         {
             // First execution. Set offset to zero.
             databaseOffset = 0;
-            context.spRocksDB->put(Utils::getCompactTimestamp(std::time(nullptr)), "0");
+            context.spRocksDB->put(
+                Utils::getCompactTimestamp(std::time(nullptr)), "0", Components::Columns::CURRENT_OFFSET);
         }
 
         return databaseOffset;
     }
 
+    /**
+     * @brief Reads and returns the file hash from the database.
+     *
+     * @param context Updater context configured with the database driver.
+     * @return unsigned int Last file hash from the database.
+     */
+    std::string getDatabaseFileHash(const UpdaterBaseContext& context) const
+    {
+        try
+        {
+            return context.spRocksDB->getLastKeyValue(Components::Columns::DOWNLOADED_FILE_HASH).second.ToString();
+        }
+        catch (const std::runtime_error& e)
```

Comment:
Isn't it possible to specify the exception to make sure to enter here only on the first run?

Reply:
Do you mean to modify the wrapper?
The `getLastKeyValue` method throws a runtime_error when there is no data in the column.

![image](https://github.com/wazuh/wazuh/assets/42682247/de86fd41-80dd-4084-a84e-bbb1f178a96f)

IMO, and for sure outside this issue, the wrapper shouldn't return an exception when no data is found, maybe an `std::optional` would be better.

Reply:
I understand, there is not much to do...